### ISSUE TITLE
perf2bolt crashes

### ISSUE THREAD

**Issue by kmod** [facebookincubator/BOLT#187]
This could certainly be user error due to the weird situation but I'm getting some crashes with perf2bolt.

My steps to create these files were:
- On my Ubuntu 20.04 machine, create a 16.04 docker container
- Use the 16.04 `perf` binary from an older kernel, but with the running newer kernel
- Run the profiling run
- Run perf2bolt, it crashes

Things I tried:
- I passed `-jump-tables=none` to the perf2bolt command, and it crashed at a different point
- I ran perf2bolt outside of the docker container (with a perf binary matching the kernel version) and it had the same crash
- I fed the .perf file directly to bolt with the `-p` flag, and that surprisingly worked

Here's the .perf file and .so file that it crashes on:
https://www.dropbox.com/s/39yagoome8hf70f/perf2bolt_crash.tar.gz?dl=0
And you should be able to reproduce the crash with `perf2bolt -p libpython3.8-pyston2.3.so.1.0.perf libpython3.8-pyston2.3.so.1.0.prebolt -o /dev/null`

**Comment by aaupov** [comment]
Hi @kmod,

Sorry for taking so long to get to this issue. I think I was able to repro the issue. Is that the assertion that you were seeing?
```
perf2bolt: /vagrant/llvm-bolt/bolt/src/BinaryBasicBlock.cpp:335: 
void llvm::bolt::BinaryBasicBlock::removePredecessor(llvm::bolt::BinaryBasicBlock*, bool): 
Assertion `Erased && "Pred is not a predecessor of this block!"' failed.
```

**Comment by kmod** [comment]
Hey sorry, it's been so long since I ran into it I don't remember the error message I got

**Comment by kmod** [comment]
Ok sorry for the delay, I reproduced the issue using a more-recent version of bolt and yes I get that error

**Comment by aaupov** [comment]
Should be resolved with 7d47004d228ab5d72b6e49000824a07415d4a10f
